### ISSUE TITLE
[2024-08-08]: Naming of parameters and derived quantities

### ISSUE THREAD

**Issue by kellijohnson-NOAA** [NOAA-FIMS/seaside-chats#10]
### Who will lead the topic?

kellijohnson-NOAA

### Topic description

Some rules need to be created for how to name parameters and derived quantities in the source code and R/json output.

### Topic background

With so many developers (current and future) we want to ensure that the naming scheme used in the source code is consistent and understandable to those outside of the project. This is not possible if guidance does not exist. The R interface group started making a list of rules and found that the more people we talked to the more those rules developed/changed because of things that we did not think of. We would like to go over our [proposed set of rules](https://docs.google.com/spreadsheets/d/1SnlXcfL90w6lEbPx1eRVRBXAEP3_97tzE1Oa_n2ivpM/edit?gid=1044110628#gid=1044110628) and finalize this list before @peterkuriyama-NOAA starts making changes to the source code next week.

We believe that it is important for the names in the source code to match the names in the R/json output to allow users to search the source code based on what they are finding in the output. Therefore, the priority is not to create names that are pretty in figures and tables. Instead of having parameter symbols as names in the code, the R output group will create converters, e.g., steepness to `h`.

### Topic outline

* Overview of current rules
* Make decisions where they are needed
* Discuss how to choose symbols to represent parameters

### Topic resources

* [excel spreadsheet with rules](https://docs.google.com/spreadsheets/d/1SnlXcfL90w6lEbPx1eRVRBXAEP3_97tzE1Oa_n2ivpM/edit?gid=1044110628#gid=1044110628)
* [GitHub for fishdictionary](https://github.com/nmfs-fish-tools/fishdictionary)

**Comment by kellijohnson-NOAA** [comment]
Thank you to @nathanvaughan-NOAA, @JaneSullivan-NOAA, and @k-doering-NOAA for attending today's seaside chat. We further realized that things are complicated 🤕 but at the end of the day if things make sense to our majority then they will probably make sense to the majority of people that touch the code in the future. We made some forward progress, and @rklasky will be happy in that we decided season should not be abbreviated to seas 😄. In fact, we are going to go forward with not shortening any of the words that make up parameter names. For example, 🛑 with rec and continue with recruitment ✅. It becomes more complicated for derived quantities, particularly with respect to those related to management.

@peterkuriyama-NOAA a major change, not with the way things will be named but in the timing of the changes to names, is that there are too many branches out right now to work on this next week or the week after. Instead, we will need to wait until we near the end of milestone 2.
